src/main/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactory.java

```diff
@@ -383,157 +106,159 @@ private static void removeReferences(TrackSsrcs trackSsrcs, List<SourcePacketExt
+                for (RTPEncodingDesc encoding : track.getRTPEncodings())
+                {
+                    if (encoding.getPrimarySSRC() == primarySSRC)
+                    {
+                        encoding.addSecondarySsrc(
+                                secondarySSRC,
+                                semanticsMap.get(sourceGroup.getSemantics()));
+                    }
+                }
+            }
+	}
+	
+        // Process the free ssrcs
+        if (sources != null && !sources.isEmpty())
         {
-            this.secondarySsrcs = secondarySsrcs;
+            for (SourcePacketExtension source : sources)
+            {
+                long primarySSRC = source.getSSRC();
+                addPlainTrack(primarySSRC);
+            }
         }
 
-        @Override
-        public Iterator<SecondarySsrc> iterator()
-        {
-            return secondarySsrcs.iterator();
-        }
+        Collection<MediaStreamTrackDesc> tracks = tracksMap.values();
+	return tracks.toArray(new MediaStreamTrackDesc[tracks.size()]);
     }
 
-    /**
-     * Describes one or more primary video ssrcs for a single
-     * {@link MediaStreamTrackDesc}
-     */
-    private static class TrackSsrcs
-        implements Iterable<Long>
+    private MediaStreamTrackDesc addPlainTrack(long primarySSRC)
     {
-        private List<Long> trackSsrcs;
+        MediaStreamTrackDesc track = tracksMap.get(primarySSRC);
 
-        public TrackSsrcs(List<Long> trackSsrcs)
+        // Check if we already have a track for this SSRC.
+        if (track == null)
         {
-            this.trackSsrcs = trackSsrcs;
+            // It looks like this is the first time we see this SSRC,
+            // so we create a new track for it.
+            track = new MediaStreamTrackDesc(trackReceiver, false);
+            track.setRTPEncodings(
+                    new RTPEncodingDesc(track, primarySSRC));
+            tracksMap.put(primarySSRC, track);
         }
 
-        public boolean contains(Long ssrc)
-        {
-            return trackSsrcs.contains(ssrc);
-        }
+        return track;
+    }
 
-        public int size()
-        {
-            return trackSsrcs.size();
-        }
+    private void addVP8SimulcastTrack(SourceGroupPacketExtension simGroup)
+    {
+        MediaStreamTrackDesc track
+            = new MediaStreamTrackDesc(trackReceiver, true);
 
-        public Long get(int index)
-        {
-            return trackSsrcs.get(index);
-        }
+        List<RTPEncodingDesc> encodings = new ArrayList<>();
 
-        @Override
-        public Iterator<Long> iterator()
+        int qot = 0;
+        int height = VP8_SIMULCAST_BASE_LAYER_HEIGHT;
+        for (SourcePacketExtension source : simGroup.getSources())
         {
-            return trackSsrcs.iterator();
-        }
-    }
+            double frameRate = 7.5;
+            long primarySSRC = source.getSSRC();
+            tracksMap.put(primarySSRC, track);
```

Comment:
Is this intentionally inside the loop? I don't think it's a problem, but seems strange.